indexer/validate_and_format_contacts.cpp

```diff
@@ -406,4 +416,97 @@ bool ValidateLinePage(string const & page)
   return (domain == "line.me" || strings::EndsWith(domain, ".line.me"));
 }
 
+bool isSocialContactTag(string_view tag)
+{
+  return tag == "contact:instagram" || tag == "contact:facebook" || tag == "contact:twitter"
+         || tag == "contact:vk" || tag == "contact:line";
+}
+
+bool isSocialContactTag(MapObject::MetadataID const metaID)
+{
+  return metaID == MapObject::MetadataID::FMD_CONTACT_INSTAGRAM ||
+         metaID == MapObject::MetadataID::FMD_CONTACT_FACEBOOK ||
+         metaID == MapObject::MetadataID::FMD_CONTACT_TWITTER ||
+         metaID == MapObject::MetadataID::FMD_CONTACT_VK ||
+         metaID == MapObject::MetadataID::FMD_CONTACT_LINE;
+}
+
+// Functions ValidateAndFormat_{facebook,instagram,twitter,vk}(...) by default strip domain name
+// from OSM data and user input. This function prepends domain name to generate full URL.
+string socialContactToURL(string_view tag, string_view value)
+{
+  if(value.empty())
```

Comment:
```suggestion
  if (value.empty())
```

Is it a normal case? Maybe `ASSERT(!value.empty(), ());` could be more helpful?